imgparse/imgparse.py

```diff
@@ -218,6 +218,35 @@ def get_focal_length(image_path, exif_data=None, xmp_data=None, use_calibrated=F
         )
 
 
+@get_if_needed("exif_data", getter=get_exif_data, getter_args=["image_path"])
+@get_if_needed("xmp_data", getter=get_xmp_data, getter_args=["image_path"])
+def get_principal_point(
+    image_path, exif_data=None, xmp_data=None, use_calibrated=False
+):
+    """
+    Get the principal point (x, y) of the sensor that took the image.
+
+    :param image_path: the full path to the image
+    :param exif_data: used internally for memoization. Not necessary to supply.
+    :param xmp_data: used internally for memoization. Not necessary to supply.
+    :return: **principal_point** - a tuple of pixel coordinates of the principal point
+    :raises: ParsingError
+    """
+    try:
+        make, _ = get_make_and_model(image_path, exif_data)
+        xmp_tags = xmp.get_tags(make)
+        pt = list(map(float, str(xmp_data[xmp_tags.PRINCIPAL_POINT]).split(",")))
+        return PixelCoords(x=pt[0], y=pt[1])
+    except KeyError:
+        raise ParsingError(
+            "Couldn't find the principal point tag. Sensor might not be supported"
+        )
+    except ValueError:
+        raise ParsingError(
+            "Couldn't parse the principal point tag. Sensor might not be supported"
+        )
```

Comment:
These could be combined into one except block e.g. `except (KeyError, ValueError):`

Reply:
Done https://github.com/SenteraLLC/py-image-metadata-parser/pull/70/commits/5f918bc6a026511259eca02bd3d884fa2aec8fef